Mathlib/CategoryTheory/Functor/Derived/RightDerived.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+import Mathlib.CategoryTheory.Localization.Predicate
+
+/-!
+# Right derived functors
+
+In this file, given a functor `F : C ⥤ H`, and `L : C ⥤ D` that is a
+localization functor for `W : MorphismProperty C`, we define
+`F.totalRightDerived L W : D ⥤ H` as the left Kan extension of `F`
+along `L`: it is defined if the type class `F.HasRightDerivedFunctor W`
+asserting the existence of a left Kan extension is satisfied.
+(The name `totalRightDerived` is to avoid name-collision with
+`Functor.rightDerived` which are the right derived functors in
+the context of abelian categories.)
```

Comment:
Do you plan to redefine the abelian version using `totalRightDerived`? Maybe add a todo?

Reply:
Of course, I plan to do this! (Actually, I already have all the code doing this, but it may take a certain time to PR all of this.) The TODO is added in another PR about derivability structure https://github.com/leanprover-community/mathlib4/pull/12633/files#diff-361ff4332cd7e3dd7ca507d30f04b09322dcb207f3f5891c4811b79f850f5c7b